test/SILOptimizer/out_of_bounds_subscript_literal.swift

```diff
@@ -0,0 +1,10 @@
+// RUN: %target-swift-frontend %s -verify -emit-sil -o
+
+func test_1() -> Int {
+  let a = [1, 2, 3]
+  return a[3] // expected-error {{subscript is accessing element at index which exceeds the bounds of the array}}
```

Comment:
I would drop "subscript". Why not say: "out of bounds access to array %1 at index %0". My reason for suggesting this is the first bit of the msg that you read makes it clear what the error is "out of bounds access to an array". It also tells you what the array is and the index. (I almost want to say the length of the array as well).